device/cuda/src/seeding/counting_grid_capacities.cu

```diff
@@ -0,0 +1,98 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2021 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#include <cuda/seeding/counting_grid_capacities.hpp>
+#include <cuda/utils/definitions.hpp>
+
+namespace traccc {
+namespace cuda {
+
+__global__ void counting_grid_capacities_kernel(
+    const seedfinder_config config, sp_grid_view grid_view,
+    spacepoint_container_view spacepoints_view,
+    vecmem::data::vector_view<std::pair<unsigned int, unsigned int>>
+        sp_container_indices_view,
+    vecmem::data::vector_view<unsigned int> grid_capacities_view);
+
+void counting_grid_capacities(
+    const seedfinder_config config, sp_grid& grid,
+    host_spacepoint_container& spacepoints,
+    vecmem::vector<std::pair<unsigned int, unsigned int>>& sp_container_indices,
+    vecmem::vector<unsigned int>& grid_capacities,
+    vecmem::memory_resource& resource) {
+
+    // auto grid_view = get_data(grid, resource);
```

Comment:
@krasznaa 
`detray::get_data` standalone function for grid is not recognized in .cu extensions (but it does work with .hpp extension) - do you have any clue on this?
```
error: no instance of function template "traccc::get_data" matches the argument list
            argument types are: (traccc::sp_grid, vecmem::memory_resource)
```
Possibly it might be related to cmake stuff because it worked before the cmake reorganization.

Reply:
My educated guess is that this is the culprit:

https://github.com/acts-project/detray/blob/main/core/include/detray/grids/grid2.hpp#L441

I found that using `auto` in the template parameter list is a no-go with NVCC.

What the heck are those options meant to be anyway? :confused:

Reply:
Um that makes sense. I may have to get view_type object in host side then...
That's for whether to sort or not (which is for complete and attach populator) and size of array (which is for complete populator).

Reply:
No... We should make that `get_data(...)` function usable with NVCC. I believe it should be doable. Similar to all the changes that I started in https://github.com/acts-project/algebra-plugins/pull/39. And to which I will hopefully be able to get back to, after the craziness that is this week...

Reply:
That's doable if i unpack the pack. Problem is that populators will have redundant template parameters :<